manifest/ocischema/manifest.go

```diff
@@ -124,6 +127,18 @@ func (m DeserializedManifest) Payload() (string, []byte, error) {
 	return v1.MediaTypeImageManifest, m.canonical, nil
 }
 
+// Subject returns a pointer to the subject of this manifest or nil if there is
+// none
+func (m *DeserializedManifest) Subject() *distribution.Descriptor {
+	return m.Manifest.Subject
+}
+
+// Type returns empty string because image manifests with subjects cannot have
+// an artifact type.
+func (m *DeserializedManifest) Type() string {
+	return ""
+}
```

Comment:
This is wrong, the referrer type, or artifact type is the `mediaType` of the `config` blob.

Reply:
Or it might be the `artifactType` property, watch this space: https://github.com/opencontainers/image-spec/pull/1043